javascript/test/parser_test.js

```diff
@@ -47,13 +45,12 @@ describe('Parser', function () {
       keyword: 'Feature',
       name: 'hello again',
       description: undefined,
-      background: undefined,
       scenarioDefinitions: [],
       comments: []
     });
   });
 
-  it("parses feature after parse error", function () {
+  it.only("parses feature after parse error", function () {
```

Comment:
For the current Objectieve-c implementation; It is safe to execute methods
on nil (aka null) values by default; so it should be fine.
Also, the current implementation of Cucumberish executes backgrounds in the
same loop as the scenario and scenario outline already.

As a quick attempt, I have changed the parent of GHBackground to be
GHScenarioDefinition and seems to be fine; however, I need to do more test
for this.

So to conclude, it will not require too much work to achieve what is
mentioned in the main purpose.


Best Regards,
Ahmed Ali
Software Engineer specialized in Mobile Development
Phone: +31 062 0991 184

Email: eng.ahmed.ali.awad@gmail.com
LinkedIn: http://eg.linkedin.com/in/engahmedali

On Sun, Mar 13, 2016 at 11:34 PM, Aslak Hellesøy <notifications@github.com>
wrote:

> In javascript/test/parser_test.js
> <https://github.com/cucumber/gherkin/pull/174#discussion_r55945077>:
>
> >        scenarioDefinitions: [],
> >        comments: []
> >      });
> >    });
> >
> > -  it("parses feature after parse error", function () {
> > +  it.only("parses feature after parse error", function () {
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cucumber/gherkin/pull/174/files#r55945077>.
>